File: Day02/Lab03.js
let numbers = [1, 2, 3, 4, 5, 6, 30, 56, 84, 23, 100];

// Print all the numbers
numbers.forEach(it => console.log(it));

// Print the sum of all numbers
const sum = numbers.reduce((currentValue, nextValue) => currentValue + nextValue);
console.log("Sum", sum);

// Print the first odd number
const firstOddNumber = numbers.find(it => it % 2 !== 0);
console.log("First odd number", firstOddNumber);

// Generate an list of squares of all numbers
const squaresOfAllNumbers = numbers.map(it => it * it);
console.log(squaresOfAllNumbers);

// Generate a list of numbers divisible by 5. 
const divBy5 = numbers.filter(it => it  % 5 == 0);
console.log("Divisible by 5", divBy5);

// Print all the even numbers
numbers
    .filter(it => it  % 2 == 0)
    .forEach(it => console.log(it));

File: Day02/Lab03.java
import java.util.*;
import java.util.stream.*;

public class Lab03 {
    public static void main(String[] args) {
        List<Integer> numbers 
            = Arrays.asList(1, 2, 3, 4, 5, 6, 30, 56, 84, 23, 100);
        
        // Print all the numbers
        numbers.forEach(System.out::println);

        // Print the sum of all numbers
        final int sum = numbers
            .stream()
            .reduce((currentValue, nextValue) -> currentValue + nextValue)
            .get();
        System.out.println("Sum " + sum);
       
        // Print the first odd number
        final int firstOddNumber = numbers
            .stream()
            .filter(it -> it % 2 != 0)
            .findFirst()
            .get();
        System.out.println(firstOddNumber);
        
        // Generate an list of squares of all numbers
        numbers
            .stream()
            .map(it -> it * it)
            .forEach(System.out::println);
        
        // Generate a list of numbers divisible by 5. 
        numbers
            .stream()    
            .filter(it -> it % 5 == 0)
            .forEach(System.out::println);

    }
}